src/test/run-make/issue-83045/Makefile

```diff
@@ -0,0 +1,28 @@
+include ../../run-make-fulldeps/tools.mk
+
+# This test case creates a situation where the crate loader would run
+# into an ICE when confronted with an invalid setup where it cannot
+# find the dependency of a direct dependency.
+#
+# The test case makes sure that the compiler produces the expected
+# error message but does not ICE immediately after.
+#
+# See https://github.com/rust-lang/rust/issues/83045
+
+# This is a platform-independent issue, no need to waste time testing
+# everywhere.
+# only-x86_64
+# only-linux
+
+all:
+	$(RUSTC) --crate-name=a --crate-type=rlib a.rs --verbose
+	$(RUSTC) --crate-name=b --crate-type=rlib --extern a=$(TMPDIR)/liba.rlib b.rs --verbose
+	# NOTE: We use $(BARE_RUSTC) below so that the compiler can't find liba.rlib
+	#       If we used $(RUSTC) the additional `-L $(TMPDIR)` option would allow rustc to
+	#       actually find the crate.
+	$(BARE_RUSTC) --out-dir $(TMPDIR) --extern b=$(TMPDIR)/libb.rlib --crate-type=rlib \
+	    --edition=2018 c.rs  &> $(TMPDIR)/output.txt || exit 0
+	# Check that we get the expected error message
+	$(CGREP) "error[E0463]" < $(TMPDIR)/output.txt
```

Comment:
You probably need backslashes for the `[` and `]`?
Might be easier to match on the message itself which is:
```
can't find crate for `a` which `b` depends on
```

Reply:
I'll just match on `E0463`. That should be sufficient.